zingo-testutils/src/chain_generic_tests.rs

```diff
@@ -128,6 +129,92 @@ pub mod fixtures {
         assert_eq!(expected_fee, recorded_fee);
     }
 
+    /// chain-generic shielding test with dust avoidance
+    /// only shield when it would return positive value
+    pub async fn shield_positive<CC>()
+    where
+        CC: ConductChain,
+    {
+        let mut environment = CC::setup().await;
+
+        let sender = environment.fund_client_orchard(1_000_000).await;
+        let shielder = environment.create_client().await;
+
+        // creates a bunch of transparent dust outputs for recipient
+        assert_eq!(
+            with_assertions::propose_send_bump_sync_recipient(
+                &mut environment,
+                &sender,
+                &shielder,
+                vec![
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000)
+                ],
+            )
+            .await,
+            8 * MARGINAL_FEE.into_u64()
+        );
+
+        // the recipient cannot propose shielding
+        assertions::assert_cant_shield(&shielder).await;
+
+        // creates an output for recipient which is too small too send, but not dust
+        let nondust_1 = 6_000u64;
+        assert_eq!(
+            with_assertions::propose_send_bump_sync_recipient(
+                &mut environment,
+                &sender,
+                &shielder,
+                vec![(Transparent, nondust_1)],
+            )
+            .await,
+            3 * MARGINAL_FEE.into_u64()
+        );
+
+        // the recipient cannot propose shielding
+        assertions::assert_cant_shield(&shielder).await;
+
+        // creates an output for recipient which combined with the previous is sendable, but would create dust: still not enough to be worth a shield
+        let nondust_2 = 14_000u64;
```

Comment:
maybe im getting this wrong but if all dust is ignored, non dust notes sum = 6_000 + 14_000 = 20_000
logical actions = 2 for 2 t-note spends plus 1 for orchard action = 3
fee = 3 * 5_000 = 15_000
received note value = 20_000 - 15_000 = 5_000 which is *not* dust
unless im missing something can we change the values to make the note 3_000 or something?

also a caveat is the orchard actions may pad to 2 which makes the received note value 0, still need to clarify this as I haven't found this in the proposal code but in testing this seems to occur